jdaviz/configs/default/plugins/collapse/collapse.py

```diff
@@ -130,35 +132,44 @@ def vue_collapse(self, *args, **kwargs):
         if self.selected_axis == 0:
             spec_min = float(self.spectral_min) * u.Unit(self.spectral_unit)
             spec_max = float(self.spectral_max) * u.Unit(self.spectral_unit)
-            spec = spec.spectral_slab(spec_min, spec_max)
-
-        collapsed_spec = getattr(spec, self.selected_func.lower())(
-            axis=self.selected_axis)
-
-        data = Data(coords=collapsed_spec.wcs)
-        data['flux'] = collapsed_spec.filled_data[...]
+            spec = spectral_slab(spec, spec_min, spec_max)
+
+        # FIXME: Does not work.
+        # collapsed_spec = spec.collapse(self.selected_func.lower(), axis=self.selected_axis)
+        # data = Data(coords=collapsed_spec.wcs)
```

Comment:
I am not sure how to use `specutils.Spectrum1D.collapse` to get spatial-spectral object correctly here. Particularly, the axis order might not be the same as `spectral-cube`; not sure, as the warnings about flipping axis from `specutils` is very confusing.

But more importantly, do we really need so many options for axis if people only ever wants to collapse to get spatial-spatial?

Reply:
In offline discussion we decided that giving "spectral" and "spatial" collapse options rather than numerical axes is the way to go here. None of us could think of a use case for collapsing only one spatial axis. FYI the specutils collapse method should take those two strings as valid `axis` inputs, so it is hopefully easy to implement.

Reply:
As discussed in 2021-12-16 tag-up, "spectral" collapse already happens automatically on load, so having it also in the plugin would be redundant.

Reply:
I forgot what's the conclusion here. Do we still support "spectral" collapse or just get rid of it?

Reply:
I would say leave it in :shrug:

Reply:
I cannot make sense of the previous implementation using spectral-cube, so I removed "axis" option altogether. #yolo